boa_engine/src/builtins/eval/mod.rs

```diff
@@ -228,7 +228,7 @@ impl Eval {
         let mut var_env = var_environment.compile_env();
 
         let mut compiler = ByteCompiler::new(
-            Sym::MAIN,
+            Sym::MAIN.to_js_string(context.interner()),
```

Comment:
Is there a reason to not use the `js_string!` macro directly for staticly known strings? For example in this case `js_string!("<main>")`.

Same goes for some other places in the PR where this is the case.